pkgs/development/libraries/nss/default.nix

```
@@ -52,43 +58,33 @@ in stdenv.mkDerivation rec {
 
   preConfigure = "cd nss";
 
-  makeFlags = let
-    # NSS's build systems expects aarch32 to be called arm; if we pass in armv6l/armv7l, it
-    # fails with a linker error
-    cpu = if stdenv.hostPlatform.isAarch32 then "arm" else stdenv.hostPlatform.parsed.cpu.name;
-  in [
-    "NSPR_INCLUDE_DIR=${nspr.dev}/include"
-    "NSPR_LIB_DIR=${nspr.out}/lib"
-    "NSDISTMODE=copy"
-    "BUILD_OPT=1"
-    "SOURCE_PREFIX=\$(out)"
-    "NSS_ENABLE_ECC=1"
-    "USE_SYSTEM_ZLIB=1"
-    "NSS_USE_SYSTEM_SQLITE=1"
-    "NATIVE_CC=${buildPackages.stdenv.cc}/bin/cc"
-  ] ++ stdenv.lib.optionals (!stdenv.isDarwin) [
-    # Pass in CPU even if we're not cross compiling, because otherwise it tries to guess with
-    # uname, which can be wrong if e.g. we're compiling for aarch32 on aarch64
-    "OS_TEST=${cpu}"
-    "CPU_ARCH=${cpu}"
-  ] ++ stdenv.lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) [
-    "CROSS_COMPILE=1"
-    "NSS_DISABLE_GTESTS=1" # don't want to build tests when cross-compiling
-  ] ++ stdenv.lib.optional stdenv.is64bit "USE_64=1"
-    ++ stdenv.lib.optional stdenv.isDarwin "CCC=clang++";
-
-  NIX_CFLAGS_COMPILE = "-Wno-error";
-
-  # TODO(@oxij): investigate this: `make -n check` works but `make
-  # check` fails with "no rule", same for "installcheck".
-  doCheck = false;
-  doInstallCheck = false;
-
-  postInstall = ''
+  buildPhase = let
+    getArch = platform: if platform.isx86_64 then "x64"
+          else if platform.isx86_32 then "ia32"
+          else if platform.isAarch32 then "arm"
+          else if platform.isAarch64 then "arm64"
+          else platform.parsed.cpu.name;
+    host = getArch stdenv.hostPlatform;
+    target = getArch stdenv.buildPlatform;
+  in ''
+    sed -i 's|nss_dist_dir="$dist_dir"|nss_dist_dir="'$out'"|;s|nss_dist_obj_dir="$obj_dir"|nss_dist_obj_dir="'$out'"|' build.sh
+    ./build.sh -v --opt \
+      --with-nspr=${nspr.dev}/include:${nspr.out}/lib \
+      --system-sqlite \
+      --enable-legacy-db \
+      --target ${target} \
+      -Dhost_arch=${host} \
+      -Duse_system_zlib=1 \
+      ${stdenv.lib.optionalString stdenv.isDarwin "--clang"} \
+      ${stdenv.lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) "--disable-tests"}
+  '';
+
+  NIX_CFLAGS_COMPILE = "-Wno-error -DNIX_NSS_LIBDIR=\"$(out)/lib/\"";
```

Weird, when would it even get interpolated

```suggestion
  NIX_CFLAGS_COMPILE = "-Wno-error -DNIX_NSS_LIBDIR=\"${placeholder "out"}/lib/\"";
```

---

How about `NIX_CFLAGS_COMPILE = ''-Wno-error -DNIX_NSS_LIBDIR="$out/lib/"'';` or `NIX_CFLAGS_COMPILE = "-Wno-error -DNIX_NSS_LIBDIR=\"$out/lib/\"";`?
Nix shouldn't (and doesn't) care either way, since it's not a `${`.

---

Well that was precisely my point — Nix does not care and neither does anything else, if I am not mistaken. Only thing that interpolates `$(foo)` syntax is `make` but `NIX_CFLAGS_COMPILE` runs behind almost everything’s back so unless I am missing something, literal `$(out)` would be passed to the compiler.

---

Right, yes. The reason why I put this there is that the patch introduced by @aszlig in 2012 patches a Makefile to set this and later uses it in code. I'm honestly not sure why it is there or how to test it. The commit talks about "built-in CA root certificates", which I can't seem to find in any of the nss outputs.

---

I don't know, but our default CA certs actually do come from NSS: https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/misc/cacert/default.nix

---

Oh, right. If you moved it out of Makefile, you definitely cannot use `make`-specific interpolation and will need to switch to Nix’s `placeholder`.

---

Thanks, I did that. I had to move it out of the Makefile, because it doesn't get picked up from there anymore with their "new" (dependent on python2 -.-) build system.